cmd/access/node_builder/access_node_builder.go

```diff
@@ -1837,17 +1845,18 @@ func (builder *FlowAccessNodeBuilder) Build() (cmd.Node, error) {
 			return nil
 		}).
 		Module("processed finalized block height consumer progress", func(node *cmd.NodeConfig) error {
-			processedFinalizedBlockHeight = bstorage.NewConsumerProgress(builder.DB, module.ConsumeProgressIngestionEngineBlockHeight)
+			processedFinalizedBlockHeight = store.NewConsumerProgress(badgerimpl.ToDB(builder.DB), module.ConsumeProgressIngestionEngineBlockHeight)
 			return nil
 		}).
 		Module("processed last full block height monotonic consumer progress", func(node *cmd.NodeConfig) error {
 			rootBlockHeight := node.State.Params().FinalizedRoot().Height
 
-			var err error
-			lastFullBlockHeight, err = counters.NewPersistentStrictMonotonicCounter(
-				bstorage.NewConsumerProgress(builder.DB, module.ConsumeProgressLastFullBlockHeight),
-				rootBlockHeight,
-			)
+			progress, err := store.NewConsumerProgress(badgerimpl.ToDB(builder.DB), module.ConsumeProgressLastFullBlockHeight).Initialize(rootBlockHeight)
+			if err != nil {
+				return err
+			}
+
+			lastFullBlockHeight, err = counters.NewPersistentStrictMonotonicCounter(progress)
```

Comment:
`NewPersistentStrictMonotonicCounter` now only takes a consumer progress that has been initialized. The initialization steps is moved out of the `PersistentStrictMonotonicCounter`, so that the initialization logic can be reused by other modules, and forces the caller to initialize the progress before being used by the counter. 

It also simplifies the counter, as it doesn't need to worry about whether the progress has been initialized or not, since it must have been initialized.